### PR TITLE
Feat/add bike estimator

### PR DESCRIPTION
**Pull request by Memphisd**
### Description
Implements the BIKE estimator, including two basic direct attacks using the SDEstimator (key and message)

### Review process
check conformity with general framework, overall sanity, coding styles, tests etc.

### CODE REVIEW

**Review by Dioprz — CHANGES_REQUESTED** [review]
Without looking at the math, it looks good to me. Just some minor fixes in the headers and possibly in the docstrings.

File: cryptographic_estimators/BIKEEstimator/BIKEAlgorithms/sd_key_attack.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        """
+        Return the time complexity of the algorithm for a given set of parameters
+
+        INPUT:
+
+        - ``parameters`` -- dictionary including the parameters
+
+        """
+        return self.get_fastest_sd_algorithm().memory_complexity()
+
```
[the remark this answers is not in the document]
Comment (Memphisd) [reply cryptographic_estimators/BIKEEstimator/BIKEAlgorithms/sd_key_attack.py]: true... don't know how that happened, removed the duplicates

[the remark this answers is not in the document]
Comment (Memphisd) [reply cryptographic_estimators/BIKEEstimator/BIKEAlgorithms/sd_key_attack.py]: For the methods that should be ok, the ```_compute_tilde_o...``` functions will only be called if the complexity type of the algorithm has been changed to "tildeO" but in that case it was propagated to the internal SDEstimator object (due to the overridden setter property). Hence the ```memory_complexity()``` and/or ```time_complexity()``` of the SDEstimator object will make the switch between the right complexities.

File: cryptographic_estimators/BIKEEstimator/bike_estimator.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        sage: A.table(show_all_parameters=1) # long time
+        +-------------+------------------+--------------------------------------------------------------------+
+        |             |                  |                              estimate                              |
+        +-------------+------------------+------+--------+----------------------------------------------------+
+        | algorithm   |   attack_type    | time | memory |                     parameters                     |
+        +-------------+------------------+------+--------+----------------------------------------------------+
+        | SDKeyAttack |   key-recovery   | 57.0 |   33.7 | {'r': 9, 'p': 2, 'l': 41, 'SD-algorithm': 'Stern'} |
+        | SDMsgAttack | message-recovery | 58.3 |   33.7 | {'r': 9, 'p': 2, 'l': 41, 'SD-algorithm': 'Stern'} |
+        +-------------+------------------+------+--------+----------------------------------------------------+
+
```
Comment (FloydZ) [inline cryptographic_estimators/BIKEEstimator/bike_estimator.py:99]: maybe we should also add the real BIKE parameters as slow tests?

**Review by FloydZ — APPROVED** [review]

File: cryptographic_estimators/BIKEEstimator/bike_estimator.py (hunk 2 above)
Comment (Memphisd) [reply cryptographic_estimators/BIKEEstimator/bike_estimator.py:99]: we can, but it takes very long, didn't want to overload the CI even more

**Comment by Memphisd** [comment]
was recreated by @AnaCaceres circumventing the Sonar checks

**Pull request closed without merging**